### ISSUE TITLE
[Question] How to pass the batch size wenn fine tuning the SAM model on own dataset?

### ISSUE THREAD

**Issue by wilhem** [facebookresearch/sam2#447]
I have a problem, when passing more than one element to the network during the training phase on my dataset.

The problem is here:

`predictor.set_image(image)`

if the image has shape: (16, 1024, 1024, 3) where 16 is the batch size, and 3 the number of channels,
then I get an error, that the dimension does not match the desired input.
If the batch size is 1, then it works.

How to deal with it?

**Comment by Akshat-Vashisht** [comment]
Could you please elaborate on what you are trying to do with code snippets and/or screenshots?

I would like to know how you are fine-tuning the model on your data to be able to offer a solution.

**Comment by wilhem** [comment]
> @wilhem Could you please elaborate on what you are trying to do with code snippets and/or screenshots?
> 
> I would like to know how you are fine-tuning the model on your data to be able to offer a solution. Based on my experience and understanding, I never used the set_image method during the fine-tuning/training process.

Then read [this](https://www.datacamp.com/tutorial/sam2-fine-tuning) tutorial please.
It answers all your questions, since it is exactly what I want to do.
Passing more than one image to set_image does not work.

**Comment by Akshat-Vashisht** [comment]
I see @wilhem 

This approach seems to not be following the official training/fine-tuning guide in the [docs](https://github.com/facebookresearch/sam2/blob/main/training/README.md). I would suggest having a look at that, it is pretty straightforward. 

Another useful resource where you get the entire process from generating masks (with ease), to training the model, to using the weights to get the new predictions is the guide provided by Roboflow. https://blog.roboflow.com/fine-tune-sam-2-1/amp/

The mask generation is made simpler with its smart mask generator tool and it outputs the dataset in a format that is already structured as per the official documentation.

Although this approach is completely different from what you were doing, it accomplishes the same purpose of fine-tuning the model on a custom dataset. I am attaching a result of fine-tuning SAM2.1 on a dataset of leaves for segmentation that I did recently following the same guide

Let me know if it helps. If not, I'll be happy to assist in any other way :D

![test3](https://github.com/user-attachments/assets/c72e4787-9e08-4701-a3e3-fa7b364dc390)

**Comment by IKnowWhoo** [comment]
Thank you so much for your reply @Akshat-Vashisht. I have read this tutorial but still have some questions, it would be so greatful if you can reply. First, how can I load my own dataset in coco format? With some reply metioned that sam2/configs/sam2.1_training/sam2.1_hiera_b+_MOSE_finetune.yaml, can I just modify this part if I convert my dataset into MOSE format? `  # PATHS to Dataset
  img_folder: null # PATH to MOSE JPEGImages folder
  gt_folder: null  # PATH to MOSE Annotations folder` Second, shold I modify rearning rate in sam2/configs/sam2.1_training/sam2.1_hiera_b+_MOSE_finetune.yaml? Thanks.

**Comment by Akshat-Vashisht** [comment]
@IKnowWhoo 

1. Yes, that should work. Make sure to follow it step by step (https://github.com/facebookresearch/sam2/blob/main/training/README.md)

It even mentions the same

> To get started with the training code, we provide a simple example to fine-tune our checkpoints on [MOSE](https://henghuiding.github.io/MOSE/) dataset, which can be extended to your custom datasets.

2. Learning rate is something that I can't comment on directly since it will vary on a case-to-case basis. I would recommend trying out the training with default LR and then running it with a slightly lower and higher LR. Observe the results and then update the LR until you are happy with the performance. It might require a little trial and error, based on your dataset, to find a "good" learning rate. However you might even be happy with the default learning rate results


Let me know if you need help with anything.
Cheers :)
